src/App/Pages/Accounts/TwoFactorPage.xaml.cs

```diff
@@ -190,7 +193,7 @@ private async Task StartDeviceApprovalOptionsAsync()
 
         private async Task TwoFactorAuthSuccessAsync()
         {
-            if (_authingWithSso)
+            if (_authingWithSso && await _vaultTimeoutService.IsLockedAsync())
```

Comment:
♻️ I don't like this kind of logic being in the Page, could we refactor this method and move the conditions to the ViewModel and from there decide if we need to go to the Lock page or the tabs with the alternate main page?
Maybe create two success methods here to be called from the VM actions:
`Task TwoFactorAuthSuccessWithSSOLockedAsync()`
`Task TwoFactorAuthSuccessToMainAsync()`